src/set.rs

```diff
@@ -1818,6 +1979,45 @@ mod test_set {
         assert_eq!(format!("{:?}", empty), "{}");
     }
 
+    #[test]
+    fn test_trivial_drain() {
+        let mut s = HashSet::<i32>::new();
+        for _ in s.drain() {}
+        assert!(s.is_empty());
+        drop(s);
+
+        let mut s = HashSet::<i32>::new();
+        drop(s.drain());
+        assert!(s.is_empty());
+    }
+
+    #[test]
+    fn test_drain() {
+        let mut s: HashSet<_> = (1..100).collect();
+
+        // try this a bunch of times to make sure we don't screw up internal state.
```

Comment:
I'm actually not so concerned about repeated runs, but more about if the map (or set) is in various stages of refill. That is, I want to make sure that drain doesn't do anything weird when the "rest" map is still non-empty for example. Could you add a test that specifically checks both a map that _isn't_ split and one that is? I also want to make sure that even if a `Drain` is forgotten (with `mem::forget`), nothing bad happens when the map is then dropped.

Reply:
ok will do!